lib/deaf_grandma.rb

```diff
@@ -19,9 +19,17 @@ def run!
 
 
   def speak(input)
-
-    #Implement your code here <<<<<<<<<
-
+    if input.match(/BYE/) and @bye_counter == 2
+      @bye_counter = 0
+      'SEE YOU LATER SONNY!'
+    elsif input.match(/^[^a-z]*$/)
```

Comment:
clever.  The only case I can think of where this might not give the desired result is if the input consists only of whitespace.  But that's going to be a problem for the approach I had in mind too (`input == input.upcase`).  Maybe something like: `/^[^a-z]*[A-Z][^a-z]*$/`?  Although that starts to get a lot hairier.